py-polars/src/series/import.rs

```diff
@@ -0,0 +1,128 @@
+use polars::export::arrow;
+use polars::export::arrow::array::Array;
+use polars::export::arrow::ffi;
+use polars::export::arrow::ffi::{
+    ArrowArray, ArrowArrayStream, ArrowArrayStreamReader, ArrowSchema,
+};
+use pyo3::exceptions::{PyTypeError, PyValueError};
+use pyo3::prelude::*;
+use pyo3::types::{PyCapsule, PyTuple, PyType};
+
+use super::*;
+
+/// Validate PyCapsule has provided name
+fn validate_pycapsule_name(capsule: &Bound<PyCapsule>, expected_name: &str) -> PyResult<()> {
+    let capsule_name = capsule.name()?;
+    if let Some(capsule_name) = capsule_name {
+        let capsule_name = capsule_name.to_str()?;
+        if capsule_name != expected_name {
+            return Err(PyValueError::new_err(format!(
+                "Expected name '{}' in PyCapsule, instead got '{}'",
+                expected_name, capsule_name
+            )));
+        }
+    } else {
+        return Err(PyValueError::new_err(
+            "Expected schema PyCapsule to have name set.",
+        ));
+    }
+
+    Ok(())
+}
+
+/// Import `__arrow_c_array__` across Python boundary
+pub(crate) fn call_arrow_c_array<'py>(
+    ob: &'py Bound<PyAny>,
+) -> PyResult<(Bound<'py, PyCapsule>, Bound<'py, PyCapsule>)> {
+    if !ob.hasattr("__arrow_c_array__")? {
+        return Err(PyValueError::new_err(
+            "Expected an object with dunder __arrow_c_array__",
+        ));
+    }
+
+    let tuple = ob.getattr("__arrow_c_array__")?.call0()?;
+    if !tuple.is_instance_of::<PyTuple>() {
+        return Err(PyTypeError::new_err(
+            "Expected __arrow_c_array__ to return a tuple.",
+        ));
+    }
+
+    let schema_capsule = tuple.get_item(0)?.downcast_into()?;
+    let array_capsule = tuple.get_item(1)?.downcast_into()?;
+    Ok((schema_capsule, array_capsule))
+}
+
+pub(crate) fn import_array_pycapsules(
+    schema_capsule: &Bound<PyCapsule>,
+    array_capsule: &Bound<PyCapsule>,
+) -> PyResult<(arrow::datatypes::Field, Box<dyn Array>)> {
+    validate_pycapsule_name(schema_capsule, "arrow_schema")?;
+    validate_pycapsule_name(array_capsule, "arrow_array")?;
+
```

Comment:
Can you add a `// SAFETY` comment explaining which invariants must hold here?

Reply:
Take a look and see if those safety comments are ok